src/dataExplorer/context/persistance.tsx

```diff
@@ -1,43 +1,57 @@
-import React, {FC, createContext} from 'react'
-import {TimeRange} from 'src/types'
+import React, {FC, createContext, useCallback} from 'react'
+import {TimeRange, RecursivePartial} from 'src/types'
 import {DEFAULT_TIME_RANGE} from 'src/shared/constants/timeRanges'
 import {useSessionStorage} from 'src/dataExplorer/shared/utils'
 import {Bucket} from 'src/types'
+import {isFlagEnabled} from 'src/shared/utils/featureFlag'
 
-interface MeasurementSelection {
+interface SchemaComposition {
+  synced: boolean
+  diverged: boolean
+}
```

Comment:
Great idea!

The synced thing is weird. Since I have trouble imagining the use case for sync'ing as it stands now. This is what I take it to be:
```
interface SchemaComposition {
  synced: boolean // defined == cannot modify session's schema
  diverged: boolean // defined == cannot re-sync. (e.g. user has typed in the composition block)
}
```

After convo with @juliajames12 tmrw, I'll push comments into this code.